src/pages/Chat/steps/NameAgeInput.tsx

```diff
@@ -18,86 +23,96 @@ interface Actions {
 }
 
 interface Props {
+  appear?: boolean
   onSubmit: (state?: FormValues) => void
 }
 
-const isDone = (values: FormValues) =>
-  values.firstName.length > 0 &&
-  values.lastName.length > 0 &&
-  String(values.age).length >= 2 &&
-  values.age > 0
+const handleChange = <K extends keyof Step1State>(
+  field: K,
+  chatState: ChatActions,
+) =>
+  pipe<React.ChangeEvent<HTMLInputElement>, Step1State[K], void>(
+    pathOr<string>('', ['target', 'value']),
+    (val: string | number) => chatState.setStep1Prop(field, val),
+  )
 
-export const NameAgeInput: React.SFC<Props> = ({ onSubmit }) => (
+const isDone = (
+  values: FormValues = { firstName: '', lastName: '', age: '' },
+) =>
+  pathOr(0, ['firstName', 'length'], values) > 0 &&
```

Comment:
I'm not super fond of this. Could we perhaps use something like `yup` instead?

The equivalent would be something like this:

```
const schema = yup.object({
  firstName: yup.string().required(),
  lastName: yup.string().required(),
  age: yup
    .number()
    .positive()
    .moreThan(10)
    .required(),
})

const isDone = (values) => schema.isValidSync(values)
```